src/ovirtapi/transform.js

```diff
@@ -661,10 +655,79 @@ const StorageDomain = {
   toInternal ({ storageDomain }: { storageDomain: ApiStorageDomainType }): StorageDomainType {
     const permissions = Permissions.toInternal({ permissions: storageDomain?.permissions?.permission })
 
+    // frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/dataprovider/AsyncDataProvider.java#getDiskVolumeFormat
+    const diskAttributesForDiskType: {
+      ['thin' | 'pre']: { sparse: boolean, format?: 'raw' | 'cow' }
+    } = {
+      thin: { sparse: true },
+      pre: { sparse: false },
+    }
+
+    // frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/dataprovider/AsyncDataProvider.java#getVolumeType
+    const defaultDiskFormatToSparse: {
+      ['raw' | 'cow']: boolean
+    } = {
+      cow: true,
+      raw: true,
+    }
+
+    const storageType = storageDomain?.storage?.type ?? 'unknown'
+    let storageSubType
+    switch (storageType) {
+      case 'nfs':
+      case 'localfs':
+      case 'posixfs':
+      case 'glusterfs':
+      case 'glance':
+        storageSubType = 'file'
+        diskAttributesForDiskType.thin.format = 'raw'
+        diskAttributesForDiskType.pre.format = 'raw'
+        defaultDiskFormatToSparse.cow = true
+        defaultDiskFormatToSparse.raw = true
+        break
+
+      case 'fcp':
+      case 'iscsi':
+        storageSubType = 'block'
+        diskAttributesForDiskType.thin.format = 'cow'
+        diskAttributesForDiskType.pre.format = 'raw'
+        defaultDiskFormatToSparse.cow = true
+        defaultDiskFormatToSparse.raw = false
+        break
+
+      case 'cinder':
+      case 'managed_block_storage':
+        storageSubType = 'openstack'
+        diskAttributesForDiskType.thin.format = undefined
+        diskAttributesForDiskType.pre.format = 'raw'
+        defaultDiskFormatToSparse.cow = true
+        defaultDiskFormatToSparse.raw = false
+        break
+
+      case 'unmanaged':
+        storageSubType = 'kubernetes'
+        diskAttributesForDiskType.thin.format = undefined
```

Comment:
Format is optional for disk and explicitly set to undefined here. However we use `format` as a key to retrieve i.e. from ` defaultDiskFormatToSparse`.  In Webadmin `VolumeFormat.Unassigned` is used in such cases. Perhaps we should add it as well?

Reply:
REST api `DiskFormat` of `undefined` will map to a `VolumeFormat.Unassigned`

Reply:
OK. Please write it as a comment in the code.

Reply:
> OK. Please write it as a comment in the code.

To be clear - this comment will be about the internal type mappings that happen between REST and the BLL.  I'm not even sure if it makes sense to make that comment.